### PR TITLE
fix a couple of python warnings

### PR DESCRIPTION
**Pull request by nirvn**
@m-kuhn , @nyalldawson , a couple of tweaks to reduce the noise output in QGIS' console.

### CODE REVIEW

File: python/plugins/MetaSearch/util.py [hunk 1]
```diff
@@ -61,7 +62,9 @@ def __init__(self):
         """init"""
         self.ppath = os.path.dirname(os.path.abspath(__file__))
         self.metadata = configparser.ConfigParser()
-        self.metadata.readfp(open(os.path.join(self.ppath, 'metadata.txt')))
+        f = codecs.open(os.path.join(self.ppath, 'metadata.txt'), "r", "utf8")
```
Comment (m-kuhn) [inline python/plugins/MetaSearch/util.py]: For extra points use

    with codecs.open(filename) as f:
        self.metadata.read_file(f)

**Review by m-kuhn — CHANGES_REQUESTED** [review]

File: python/plugins/MetaSearch/util.py (hunk 1 above)
Comment (nirvn) [inline python/plugins/MetaSearch/util.py]: @m-kuhn , what advantage(s) does to above offer over the current code? A protection against failure to open the filename?

Comment (m-kuhn) [inline python/plugins/MetaSearch/util.py]: it will close the file properly regardless of any exceptions happening.

Comment (nirvn) [inline python/plugins/MetaSearch/util.py]: @m-kuhn , ok. So it'd be:

```
with codecs.open(filename) as f:
    self.metadata.read_file(f)
    f.close()
```
or
```
with codecs.open(filename) as f:
    self.metadata.read_file(f)
f.close()
```
?

Comment (m-kuhn) [inline python/plugins/MetaSearch/util.py]: None of them. Close happens implicitly when the with block scopes out.

**Comment by nirvn** [comment]
@m-kuhn , done. Thanks for the TIL moment 😄

**Comment by m-kuhn** [comment]
Thanks @nirvn , always a pleasure :)

**Pull request merged**